search-books/src/main/java/nix/education/service/query/SolrQueryBuilderService.java

```diff
@@ -0,0 +1,80 @@
+package nix.education.service.query;
+
+import java.util.Date;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.StringJoiner;
+import lombok.SneakyThrows;
+import nix.education.util.FacetFields;
+import nix.education.util.SymbolConstants;
+import nix.education.util.SymbolsSolrQuery;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import static nix.education.util.DateFormat.DATE_FORMAT;
+
+@Service
+public class SolrQueryBuilderService {
+
+    @Value("${search.term-facets}")
+    private Set<String> termFacets;
+
+    @Value("${search.range-facets}")
+    private Set<String> rangeFacet;
+
+    @Value("#{${search.search-fields}}")
+    private Map<String, Float> fields;
+
+    private final Date START_SEARCH_BOOK_DATE;
+
+    @SneakyThrows
+    public SolrQueryBuilderService() {
+        this.START_SEARCH_BOOK_DATE = DATE_FORMAT.parse("1899-12-31 22:00:00");
```

Comment:
It's better to use this value as a constant

Reply:
ok. I'll put it to class with dates.